pyracf/common/logger.py

```diff
@@ -93,33 +96,78 @@ def log_debug(self, header_message: str, message: str) -> None:
         )
         print(f"{header}\n{message}")
 
-    def redact_strings(
+    def __redact_request_dictionary(
         self,
-        string: Union[str, bytes],
-        redact_strings: List[str],
-    ) -> str:
-        """Redact a list of strings or sequences of bytes in a string or bytes object"""
-        if not redact_strings:
-            return string
-        for to_redact in redact_strings:
-            if isinstance(string, bytes):
-                string = self.redact_string(
-                    string, redact_string=bytes(to_redact, "utf-8")
-                )
-            else:
-                string = self.redact_string(string, redact_string=to_redact)
-        return string
+        dictionary: dict,
+        secret_traits: dict,
+    ) -> dict:
+        """
+        Redact a list of specified secret traits in a request dictionary.
+        While this flow is technically used in extract functions, there are no traits to redact.
+        """
+        for trait in secret_traits:
+            segment = trait.split(":")[0]
+            if not dictionary.get(segment, {}).get(trait, {}).get("value", None):
+                continue
+            dictionary[segment][trait]["value"] = "********"
+        return dictionary
+
+    def __redact_string(
+        self,
+        input_string: Union[str, bytes],
```

Comment:
I think this should just be `input_string: str` because `__redact_string` never gets a `bytes` object from any of its callers and I actually am not sure if this would work correctly with bytes objects.